h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -262,6 +264,29 @@ public boolean onExceptionalCompletion(Throwable ex, CountedCompleter caller) {
     }
 
     public abstract void computeImpl();
+    
+    public void computeEffectiveParameters() {
+      M model = _result.get();
+      if (model != null) {
+        model._effective_parms = (P) _parms.clone();
+        model.initEffectiveParam();
+        checkEffectiveParmsDoesNotContainAuto(model._effective_parms);
+      }
+    }
+  }
+
+  public void checkEffectiveParmsDoesNotContainAuto(Model.Parameters effectiveParameters){
+    try {
+      for (Field field : effectiveParameters.getClass().getFields()) {
+        Class type = field.getType();
+        Object value = field.get(effectiveParameters);
+        if (value != null){
+          assert(!"AUTO".equalsIgnoreCase(value.toString())) : "Found AUTO value in effective parameters: " + field.getName();
+        }
+      }
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException(e);
```

Comment:
I would add message exaplaining when the error happened "Error while checking params for auto values"

Reply:
added